### PR TITLE
Add option to disable self-update

### PR DESCRIPTION
**Pull request by bboreham**
Self-update is currently broken if the agent is installed via Helm #303, and on Kubernetes pre-1.9 #300, so allow it to be turned off.

### CODE REVIEW

**Comment by cristian-radu** [comment]
LGTM

File: agent/main.go [hunk 1]
```diff
@@ -133,51 +134,53 @@ func agentManifestURL(cfg *agentConfig) string {
 }
 
 func updateAgents(cfg *agentConfig, cancel <-chan interface{}) {
-	// Self-update
-	agentPollURL := agentManifestURL(cfg)
-	log.Info("Updating self from ", agentPollURL)
+	if cfg.SelfUpdate {
```
Comment (errordeveloper) [inline agent/main.go]: nit: `if !cfg.SelfUpdate { return }` would make diff simpler

Comment (bboreham) [reply agent/main.go]: But that is not the intended behaviour - it's still supposed to update all the other stuff if you turn off self-update.

Comment (bboreham) [reply agent/main.go]: The diff looks simpler if you click on 'Hide whitespace changes'

**Review by errordeveloper — APPROVED** [review]

**Comment by bboreham** [comment]
Obsolete - closing.

**Pull request closed without merging**